### PR TITLE
fix: test flakiness

### PR DESCRIPTION
**Pull request by juliusmarminge**
while debugging #841 Tim and I discovered the tests were a bit flaky.

this brings the fixes we incorporated over there as it's own PR

### CODE REVIEW

File: packages/uploadthing/test/__test-helpers.ts [hunk 1]
```diff
@@ -36,14 +34,6 @@ export const baseHeaders = {
   "x-uploadthing-package": "vitest",
 };
 
-export const genPort = (test: Test) => {
```
Comment (juliusmarminge) [inline packages/uploadthing/test/__test-helpers.ts:39]: `server.listen(0)` finds an open port itself so we don't need this shenanigans

File: packages/uploadthing/test/__test-helpers.ts [hunk 2]
```diff
@@ -211,7 +201,7 @@ export const it = itBase.extend({
 
           // Simulate polling - at least once
           yield HttpResponse.json({ status: "still waiting" });
-          if (!file) {
+          while (!file) {
```
Comment (juliusmarminge) [inline packages/uploadthing/test/__test-helpers.ts:204]: polling can fail after first attempt, keep retrying until we have a file in db (or test times outs and fails)

File: packages/uploadthing/test/client.test.ts [hunk 3]
```diff
@@ -48,20 +48,24 @@ export const setupUTServer = (task: Test) => {
     }),
   );
 
-  const port = genPort(task);
-  const server = app.listen(port);
+  const server = app.listen(0);
+  const port = (server.address() as AddressInfo).port;
+  await new Promise((res) => server.once("listening", res));
```
Comment (juliusmarminge) [inline packages/uploadthing/test/client.test.ts:53]: server was sometimes not started before the 1st test executed leading to some occasional `Fetch Failed`'s

File: packages/uploadthing/test/client.test.ts [hunk 4]
```diff
@@ -48,20 +48,24 @@ export const setupUTServer = (task: Test) => {
...
+    uploadFiles,
+    close: () => new Promise((res) => server.close(res)),
+  };
 };
 
 describe("uploadFiles", () => {
-  it("uploads with presigned post", async ({ db, task }) => {
-    const { uploadFiles, close } = setupUTServer(task);
+  it("uploads with presigned post", async ({ db }) => {
+    const { uploadFiles, close } = await setupUTServer();
```
Comment (markflorkowski) [inline packages/uploadthing/test/client.test.ts:68]: Didn't we have a lint rule for no floating promises?

**Review by markflorkowski — APPROVED** [review]

File: packages/uploadthing/test/client.test.ts (hunk 4 above)
Comment (juliusmarminge) [reply packages/uploadthing/test/client.test.ts:68]: it wasn't a promise before. nodejs callback api :)

**Pull request merged**